### PR TITLE
feat(router): add `click_to_pay` block in payments sessions response if enabled

### PR DESCRIPTION
**Pull request by sai-harsha-vardhan**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

add `click_to_pay` block in payments sessions response if enabled

### Additional Changes

- [x] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

File: crates/router/src/core/payments.rs [hunk 1]
```diff
@@ -3364,6 +3379,64 @@ where
...
+    key_store: &domain::MerchantKeyStore,
+    authentication_product_ids: serde_json::Value,
+    payment_intent: &hyperswitch_domain_models::payments::PaymentIntent,
+) -> RouterResult<api_models::payments::SessionToken> {
+    use common_utils::id_type::MerchantConnectorAccountId;
+    use hyperswitch_domain_models::payments::ClickToPayMetaData;
+
+    use crate::consts::CLICK_TO_PAY;
+
+    let mca_ids: HashMap<String, MerchantConnectorAccountId> = authentication_product_ids
```
Comment (Narayanbhat166) [inline crates/router/src/core/payments.rs:3395]: we can have these as a type in the database column right? or atleast in the domain models

File: crates/router/src/core/payments.rs [hunk 2]
```diff
@@ -3364,6 +3379,64 @@ where
...
+            key_manager_state,
+            merchant_id,
+            click_to_pay_mca_id,
+            key_store,
+        )
+        .await
+        .to_not_found_response(errors::ApiErrorResponse::MerchantConnectorAccountNotFound {
+            id: click_to_pay_mca_id.get_string_repr().to_string(),
+        })?;
+    let click_to_pay_metadata: ClickToPayMetaData = merchant_connector_account
```
Comment (Narayanbhat166) [inline crates/router/src/core/payments.rs:3416]: are we validating this metadata when it is being stored in the database?

Comment (sai-harsha-vardhan) [reply crates/router/src/core/payments.rs:3416]: Yes, it's in the PR where the CTP_Mastercard connector is added

File: crates/router/src/core/payments.rs (hunk 1 above)
Comment (sai-harsha-vardhan) [reply crates/router/src/core/payments.rs:3395]: Will do it in a separate PR

**Review by Narayanbhat166 — APPROVED** [review]

File: crates/api_models/src/payments.rs [hunk 3]
```diff
@@ -6989,6 +6991,23 @@ pub struct ExtendedCardInfoResponse {
     pub payload: String,
 }
 
+#[derive(Debug, Clone, Eq, PartialEq, serde::Serialize, ToSchema)]
+#[serde(rename_all = "camelCase")]
+pub struct ClickToPaySessionResponse {
+    pub dpa_id: String,
+    pub dpa_name: String,
+    pub locale: String,
+    pub card_brands: Vec<String>,
```
[the remark this answers is not in the document]
Comment (sai-harsha-vardhan) [reply crates/api_models/src/payments.rs:7000]: Will do it in a separate PR

**Review by ShankarSinghC — APPROVED** [review]

**Pull request merged**